dataplicity/m2m/wsclient.py

```diff
@@ -199,7 +199,27 @@ def closed(self, code, reason=None):
         try:
             self.on_close(self)
         except Exception as error:
-            log.error('error on WSApp.closed: %s', error)
+            log.error('WSApp.closed: %s', error)
+
+    def unhandled_error(self, error):
```

Comment:
This handler gets called when ws4py gets a socket error it doesn't know about it.